server/api/classyear/classyear.controller.js

```diff
@@ -220,5 +222,20 @@ function generateCode(codeLength){
       var character = (Math.floor(Math.random() * characterOptions.length));
       code = code.concat(characterOptions[character.toString()]);
   }
+  ClassYear.findOne({"dayCodes.code":code})
```

Comment:
@agundy the reason that I put the checking inside generateCode was meant to make sure when generateCode return a code to daycode, the returning code must not be a duplicate one.